docs/ml-features.md

```diff
@@ -1389,3 +1389,145 @@ print(output.select("features", "clicked").first())
 
 # Feature Selectors
 
+## VectorSlicer
+
+`VectorSlicer` is a transformer that takes a feature vector and outputs a new feature vector with a sub-array of the original features. It is useful for extracting features from a vector column.
+
+`VectorSlicer` accepts a vector column with a specified indices, then outputs a new vector column whose values are selected via those indices. There are two types of indices, 
+
+ 1. Integer indices that represents the real indices in the vector, `setIndices()`;
+
+ 2. String indices that represents the names of features in the vector, `setNames()`.
+
+Specify by integer and string are both acceptable, moreover, you can use integer index and string name simultaneously. At least one feature must be selected. Duplicate features are not allowed, so there can be no overlap between selected indices and names. Note that if names of features are selected, an exception will be threw out when encountering with empty input attributes.
```

Comment:
nit: ***Specification*** by integer and string are both acceptable***. M***oreover,